embuilder.py

```diff
@@ -169,6 +169,10 @@ def get_all_tasks():
   return get_system_tasks()[1] + PORTS
 
 
+def handle_port_error(message):
+  raise Exception(message)
```

Comment:
Just call `exit_with_error` here to avoid the whole try/catch thing.

The problem with the generic `except Exception as e:` is that it will catch all kinds of other exceptions, no just the ones generated via handle_port_error.  We could create a specialized exception type, but that just adds more code.